### PR TITLE
session: provide uploadData with webrequest api when available

### LINKED ISSUE #4214: Is possible to catch parameters which has been sent by post request via website (in webview)? [v0.36.3]

### PR DESCRIPTION
**Pull request by deepak1556**
Fixes #4214

### CODE REVIEW

**Comment by vially** [comment]
:+1:

File: atom/common/native_mate_converters/net_converter.cc [hunk 1]
```diff
@@ -83,3 +64,36 @@ v8::Local<v8::Value> Converter<scoped_refptr<net::X509Certificate>>::ToV8(
...
+    scoped_ptr<base::DictionaryValue> upload_data_dict(
+        new base::DictionaryValue);
+    if (reader->AsBytesReader()) {
+      const net::UploadBytesElementReader* bytes_reader =
+          reader->AsBytesReader();
+      scoped_ptr<base::Value> bytes;
+      auto buffer =
+          base::BinaryValue::CreateWithCopiedBuffer(bytes_reader->bytes(),
+                                                    bytes_reader->length());
+      bytes.reset(buffer);
```
Comment (zcbenz) [inline atom/common/native_mate_converters/net_converter.cc]: Can you do it in this way?

```c++
scoped_ptr<base::Value> bytes(
    base::BinaryValue::CreateWithCopiedBuffer(...));
```

File: docs/api/session.md [hunk 2]
```diff
@@ -332,6 +332,9 @@ is about to occur.
   * `method` String
   * `resourceType` String
   * `timestamp` Double
+  * `uploadData` Object (optional)
```
Comment (zcbenz) [inline docs/api/session.md]: It should be `Array`.

**Comment by deepak1556** [comment]
Have made the changes, thanks!

**Comment by zcbenz** [comment]
:+1:

**Pull request merged**